backbone/src/main/java/org/researchstack/backbone/task/Task.java

```diff
@@ -40,6 +46,53 @@ public Task(String identifier)
     }
 
     /**
+     * Sets the current @{@link ViewTaskActivity} for this task.
+     * This function should be called only by the {@link ViewTaskActivity} instance running
+     * at its creation time.
+     * @param activity the @{@link ViewTaskActivity} instance
+     */
+    public void setActivity(ViewTaskActivity activity){
```

Comment:
These were necessary to implement the timed walk task. I needed to access the activity as the task was determining what was being displayed. And I couldn't just switch to another activity. Also my task needs to have access to the context in order to retrieve data from sensors and other amenities. I couldn't find any way to implement these features without completely having to break the current design completely (as done in MoleMapper for example). I know it's not ideal but it was the best solution I could find.